### DISCUSSION TITLE
Base System for Ticketing/Wanted (Police)

### DISCUSSION THREAD

**Discussion by HarrisOik** [EveronLife/EveronLife#135] [Ideas]
While each server should be making this as detailed or abstract to match their playstyle/communities, I believe we should at least have some very basic functionality. For example:

- Automatic "Manslaughter/Murder" charge on kill (so other people can expand on automatic charges themselves)
- Manual "Speeding" or "Attempted Manslaughter/Murder" charge assigned by the police (so others can have a baseline to make this list as complex as they wish)
- Ticket pay request (this would require a banking system, people shouldn't be expected to carry enough to pay a ticket)
- Bounty claim reward on kill (only by police)
- A wanted list that can be accessed by police

I am leaving out the obvious as rewards to police officers on arrest (send to jail functionality) and police restraining to ticket (this requires restraining as discussed in #114).
Also, in my opinion tickets should be split between all active police officers (maybe an interchangeable option for owners to decide).
Again, this should be considered as a strong template rather than a feature for owners and communities to make it as simple or complex as they wish.

**Comment by thebonbon** [comment]
> - Automatic "Manslaughter/Murder" charge on kill (so other people can expand on automatic charges themselves)
> - Manual "Speeding" or "Attempted Manslaughter/Murder" charge assigned by the police (so others can have a baseline to make this list as complex as they wish)
> 
I don't think we should add automatic charges on kill. The only thing that would fit is a speed camera but I dont like those either.
> - Ticket pay request (this would require a banking system, people shouldn't be expected to carry enough to pay a ticket)
>
Agreed
> - Bounty claim reward on kill (only by police)
>
Strongly disagree. The focus should not be on killing (pvp) and thus not giving rewards for kills.
Its fine if a server wants to add that, but should not be in the framework imo.
> - A wanted list that can be accessed by police
>
Yes some sort of wanted list / system / records.

**Comment by jasper-one** [comment]
Thanks for your post!

I'd agree with some of this but not the automatic warrants or bounties.

If it take someone out to a field and kill them why would it automatically lodge an arrest warrant? It ruins the RP.

Why would the police get money for killing someone? Best case you get paid leave and you cant perform duties. Again, it screws with RP and insentivses KOS type behaviour. I would just not have it in.

**Comment by HarrisOik** [comment]
I agree with both of you, but you have to remember that these would be considered as templates, each community can either use and remove what they do not like. As I mentioned, consider this a template and/or example rather than a feature. If you do not like the use of automatic bounty (which I completely understand your reasoning) we should at least make an example out of a speed trap.

On the other hand, if we do want pure RP situations, technically speaking the bounty should go away when a person dies (technically the person died so they are not wanted anymore), but again this will cause situations where people will want to be killed so perhaps we should indeed scrap it.

**Reply by thebonbon** [reply]
I wouldn't add it even as template / example. I think we should try to keep the framework as far away from the old aram 3 pvp "rp".
This is the one opportunity to make it something new and better.

**Comment by dan-schwerner** [comment]
Hi Guys! Checking in for an update on this. My personal take for what it's worth, as a first iteration, allowing cops to issue a ticket and as was mentioned, being dependent on the banking system would be a great first iteration and would open the door to several RP possibilities.

Question from my end, where would the ticketed amount go? For now I'd suggest having this go straight to the cop's wallet but having a modifiable divisor in case server owners would want to reduce pay outs due to greedy cops. There is definitely potential for other systems like a central pool that would contribute to better equipment, but this would introduce a lot of unnecessary complexity to begin with in my opinion. 

Also, is there any existing/ WIP functionality that handles the following:

- handling interactions with another playable entity for a player-to-player specific menu.
- standard UI components/layouts for any dialogs presented (similar to existing slider widget used for splitting quantities).
- banking system with money transfer capabilities between players. 

Thanks!

**Reply by dan-schwerner** [reply]
Following a discussion with @thebonbon, the idea is to have a system similar to nopixel's as seen here https://h2n9.github.io/OMDW/?p=Charges. The idea would be to have a list of configurable charges on the server which should be easily modifiable by server owners. Rather than having an option on the player, a charge would be added through this system, forcing the cop to get the identity of the player through RP to issue a ticket. In an effort to keeping the utilization of this system as flexible as possible for the different communities using it, I wouldn't add too many features. 

- Cop would retrieve a player's ID through roleplay.
- Cop would need to access some sort of device (purchasable tablet / PC / vehicle / PDA) where a list of charges are added to the player at any point in the RP scenario where the cop deems fit.
- Charges would have an associated "**name**", "**recommended jail time**" and "**recommended ticket price**" which can be reduced / increased at the discretion of the cop. Jail time can be togglable in case the cop doesn't wish to send a player to prison for a felony. I would recommend having an "issue warning" toggle instead as well, as several servers incorporate warnings as part of processing especially if the player processed is new / had good RP in the situations.
- Player would be notified that he's been charged and to pay the ticket immediately. 

The above system would resolve what tends to happen most of the time with cops, where they have to have multiple monitors or alt / tab out to check current ticket prices / jailtimes which tends to result in the processed player waiting until the cop finds the needed info. I could also see this being extended to not having to pay the fine immediately or serving time immediately, allowing the kind of RP where a civ doesn't have enough money to pay a fine and can be given time to make money to come to a station and pay it, or to say bye to their friends and family before going to prison, but this can come at a later stage. Having the core architecture of actually persisting charges until fine is paid or prison time is served would have a lot of potential.
